tools/src/main/java/org/apache/kafka/trogdor/agent/AgentRestResource.java

```diff
@@ -47,20 +49,19 @@ public void setAgent(Agent myAgent) {
     @GET
     @Path("/status")
     public AgentStatusResponse getStatus() throws Throwable {
-        return new AgentStatusResponse(agent().startTimeMs());
+        return agent().status();
     }
 
-    @GET
-    @Path("/faults")
-    public AgentFaultsResponse getAgentFaults() throws Throwable {
-        return agent().faults();
+    @PUT
+    @Path("/worker/create")
+    public CreateWorkerResponse createWorker(CreateWorkerRequest req) throws Throwable {
+        return agent().createWorker(req);
     }
 
     @PUT
-    @Path("/fault")
-    public Empty putAgentFault(CreateAgentFaultRequest request) throws Throwable {
-        agent().createFault(request);
-        return Empty.INSTANCE;
+    @Path("/worker/stop")
```

Comment:
I don't think this is a really a DELETE because the worker is still there, just stopped.  It's possible to imagine doing more with it in the future.

Reply:
In which case, it could be `PUT` to `/agent/workers/<id>`?

Reply:
I agree that it would be a little more "RESTy" to send different message types to `/agent/workers/$id`.  However, it's awkward to implement.  Right now, the APIs are "strongly typed" in the sense that when you send a message to `/agent/worker/stop`, you know what the format of the message is.  To implement something more complex, where you could send multiple messages to that endpoint, would be more difficult to implement with jaxrs + jetty.  Given that this is a purely internal and testing API for now, maybe we can do that later, once we've implemented more user-visible features?

Reply:
Yes, that is fine.